syncd/scripts/syncd_init_common.sh

```diff
@@ -119,74 +119,181 @@ config_syncd_cisco_8000()
+                   if $override ;then
+                      echo "Override the config $(grep $sedline $to_file) with $line in $to_file"
+                      prop=${line:0:`expr index "$line" :`}
+                      sed -i "/$prop/d" $to_file
+                      echo "        $line" >> $to_file
+                      merged_cnt+=1
                    else
-                       sedline=${line%=*}
-                       if grep -q $sedline $file ;then
-                          echo "Keep the config $(grep $sedline $file) in $file"
-                       else
-                          echo $line >> $file
-                       fi
+                      grepline=$(grep $sedline $to_file)
+                      grepline="${grepline#"${grepline%%[![:space:]]*}"}"
+                      if [ "${grepline::1}" == '#' ];then
+                         echo "        $line" >> $to_file
+                         merged_cnt+=1
+                      else
+                         echo "Keep the config $(grep $sedline $to_file) in $to_file"
+                      fi
                    fi
-               fi
-             done < $COMMON_CONFIG_BCM
-             echo "# End of chip common properties" >> $file
-         done
-         echo "Merging $PLT_CONFIG_BCM with $COMMON_CONFIG_BCM, merge files stored in $CONFIG_BCM"
-      fi
-
-      #sync the file system
-      sync
-
-      # copy the final config.bcm and sai.profile to the shared folder for 'show tech'
-      cp -f /tmp/sai.profile /var/run/sswsyncd/
-      cp -f /tmp/*.bcm /var/run/sswsyncd/
-
-      if [ -f "/tmp/sai.profile" ]; then
-          CMD_ARGS+=" -p /tmp/sai.profile"
-      elif [ -f "/etc/sai.d/sai.profile" ]; then
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+                else
+                   echo "        $line" >> $to_file
+                   merged_cnt+=1
+                fi
+            fi
+        fi
+    done < $from_file
 
-    else
+    if [ $merged_cnt -gt 0 ]; then
+         sed -i "/# Start of/a \    global:" $to_file
+         sed -i "/# Start of/a \  0:" $to_file
+         sed -i "/# Start of/a \bcm_device:" $to_file
+         sed -i "/# Start of/a \---" $to_file
+    fi
+    echo "# End of $message" >> $to_file
+    if [ $merged_cnt -gt 0 ]; then
+       sed -i "/# End of/i \..." $to_file
+    fi
+    echo "Merged $from_file to $to_file"
+}
 
-      if [ -f "/etc/sai.d/sai.profile" ]; then
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+config_syncd_bcm()
+{
+    PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
+    PLT_CONFIG_BCM=$(find $HWSKU_DIR -name '*.bcm' -not -path "*/phy/*")
+    PLT_CONFIG_YML=$(find $HWSKU_DIR -name '*.yml')
+
+    if [ ! -z "$PLT_CONFIG_BCM" ] && [ -f $PLATFORM_DIR/common_config_support ] ; then
+       cp -f $HWSKU_DIR/*.config.bcm /tmp
```

Comment:
@geans-pin There is an issue here.
For plat form, Platform: x86_64-arista_7260cx3_64, HwSKU: Arista-7260CX3-C64.
No config.bcm file under HWSKU_DIR:
```
root@str3-7260cx3-acs-14:/usr/share/sonic/hwsku# ls -al
total 93
drwxr-xr-x 1 root root  4096 Oct 31 10:45 .
drwxr-xr-x 1 root root  4096 Oct 31 11:43 ..
drwxr-xr-x 2 root root   123 Oct 30 04:54 BALANCED
drwxr-xr-x 2 root root   123 Oct 30 04:54 RDMA-CENTRIC
drwxr-xr-x 2 root root   123 Oct 30 04:54 TCP-CENTRIC
-rw-r--r-- 1 root root   226 Feb 20  2017 buffer_ports_t0.j2
-rw-r--r-- 1 root root   226 Feb 20  2017 buffer_ports_t1.j2
-rw-r--r-- 1 root root    67 Feb 20  2017 buffers.json.j2
-rw-r--r-- 1 root root  1777 Oct 31 10:45 buffers_defaults_t0.j2
-rw-r--r-- 1 root root  1777 Feb 20  2017 buffers_defaults_t0_orig.j2
-rw-r--r-- 1 root root  1976 Oct 31 10:45 buffers_defaults_t1.j2
-rw-r--r-- 1 root root  1976 Feb 20  2017 buffers_defaults_t1_orig.j2
-rw-r--r-- 1 root root  1253 Feb 20  2017 buffers_extra_queues.j2
-rw-r--r-- 1 root root   547 Feb 20  2017 buffers_pool_sizes_t0.j2
-rw-r--r-- 1 root root   544 Feb 20  2017 buffers_pool_sizes_t1.j2
-rw-r--r-- 1 root root 39593 Feb 20  2017 config.bcm.j2
-rw-r--r-- 1 root root   549 Feb 20  2017 pg_profile_lookup.ini
-rw-r--r-- 1 root root  4084 Feb 20  2017 port_config.ini
-rw-r--r-- 1 root root 10788 Feb 20  2017 qos.json.j2
-rw-r--r-- 1 root root    67 Feb 20  2017 sai.profile
```

it only has config.bcm.j2 file, so merging won't happen for this hwsku, or other hwskus which are using config.bcm.j2, not config.bcm.

The possible solution is check the correct config.bcm file location in /etc/sai.d/sai.profile, then  read correct config.bcm

```
root@str3-7260cx3-acs-14:/usr/share/sonic/hwsku# cat sai.profile 
SAI_INIT_CONFIG_FILE=/etc/sai.d/config.bcm
SAI_NUM_ECMP_MEMBERS=64
```

Reply:
ok

Reply:
See the latest commit, I've fixed this issue to search config file from sai.profile